singer-io/tap-jira
# DE-2573 Part 1: add Boards endpoint

# Description of change
As part of the desire to see the Velocity-report in Domo, the Velocity-data needs to be received from the Jira APIs. This PR constitutes of changes made to return data from the 1st of 5 API endpoints.

There are two known Boards endpoint, in the process of creating the Greenhopper endpoint for Boards, I've also added the other (documented) Agile endpoint for Boards. Note that the Greenhopper endpoint is still used by the Velocity report in the Jira GUI.

# Manual QA steps
 1) pull the branch to your local repo.
 2) drop all cached/build/dist directories related to the Jira tap, in my case that's 
  - <repo>\tap-jira\build
  - <repo>\tap-jira\dist
  - <python library directory>\site-packages\\tap_jira-2.1.4-py3.10.egg
  3) reinstall tap-jira by executing `python setup.py install`
  4) have your personal config.json ready
  5) add the following to your catalog.json and make sure the JSON validates afterwards
```
    {
      "tap_stream_id": "boardsGreenhopper",
      "key_properties": [
        "id"
      ],
      "schema": {
        "properties": {
          "id": {
            "type": [
              "null",
              "integer"
            ]
          },
          "name": {
            "type": [
              "null",
              "string"
            ]
          },
          "canEdit": {
            "type": [
              "null",
              "boolean"
            ]
          },
          "sprintSupportEnabled": {
            "type": [
              "null",
              "boolean"
            ]
          },
          "showDaysInColumn": {
            "type": [
              "null",
              "boolean"
            ]
          },
          "kanPlanEnabled": {
            "type": [
              "null",
              "boolean"
            ]
          },
          "showEpicAsPanel": {
            "type": [
              "null",
              "boolean"
            ]
          },
          "isSimpleBoard": {
            "type": [
              "null",
              "boolean"
            ]
          },
          "savedFilterId": {
            "type": [
              "null",
              "boolean"
            ]
          },
          "oldDoneIssuesCutoff": {
            "type": [
              "null",
              "string"
            ]
          }
        },
        "type": [
          "null",
          "object"
        ]
      },
      "stream": "boards",
      "metadata": [
        {
          "breadcrumb": [],
          "metadata": {
            "selected": true,
            "table-key-properties": [
              "id"
            ]
          }
        },
        {
          "breadcrumb": [
            "properties",
            "id"
          ],
          "metadata": {
            "inclusion": "automatic"
          }
        },
        {
          "breadcrumb": [
            "properties",
            "name"
          ],
          "metadata": {
            "inclusion": "available"
          }
        },
        {
          "breadcrumb": [
            "properties",
            "canEdit"
          ],
          "metadata": {
            "inclusion": "available"
          }
        },
        {
          "breadcrumb": [
            "properties",
            "sprintSupportEnabled"
          ],
          "metadata": {
            "inclusion": "available"
          }
        },
        {
          "breadcrumb": [
            "properties",
            "showDaysInColumn"
          ],
          "metadata": {
            "inclusion": "available"
          }
        },
        {
          "breadcrumb": [
            "properties",
            "kanPlanEnabled"
          ],
          "metadata": {
            "inclusion": "available"
          }
        },
        {
          "breadcrumb": [
            "properties",
            "showEpicAsPanel"
          ],
          "metadata": {
            "inclusion": "available"
          }
        },
        {
          "breadcrumb": [
            "properties",
            "isSimpleBoard"
          ],
          "metadata": {
            "inclusion": "available"
          }
        },
        {
          "breadcrumb": [
            "properties",
            "savedFilterId"
          ],
          "metadata": {
            "inclusion": "available"
          }
        },
        {
          "breadcrumb": [
            "properties",
            "oldDoneIssuesCutoff"
          ],
          "metadata": {
            "inclusion": "available"
          }
        }
      ]
    }
````
 6) run `tap-jira --config config.json -p catalog.json > output.json` to see what tap-jira outputs. Now the boards are loaded somewhere in the beginning, so if your code has been running for over, say, 10 seconds, break (CTRL-C) and check your `output.json`
 
# Risks
 - It is said that the Greenhopper endpoints are deprecated, however, those very same endpoints are used when loading the Velocity report via the Jira GUI: https://degreedjira.atlassian.net/jira/software/projects/DE/boards/39/reports/velocity
 
# Rollback steps
 - revert this branch
